akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessImpl.scala

```diff
@@ -153,13 +156,23 @@ private[akka] final class ShardedDaemonProcessImpl(system: ActorSystem[_])
     }
   }
 
+  // Java API
   def init[T](
       messageClass: Class[T],
       name: String,
       numberOfInstances: Int,
       behaviorFactory: function.Function[Integer, Behavior[T]]): Unit =
     init(name, numberOfInstances, n => behaviorFactory(n))(ClassTag(messageClass))
 
+  override def init[T](
+      messageClass: Class[T],
+      name: String,
+      numberOfInstances: Int,
+      behaviorFactory: function.Function[Int, Behavior[T]],
+      stopMessage: T): Unit =
```

Comment:
It's good that it's not an option. 

Can we pull the `ApiMayChange` card and remove the Option from the other `init` method? We will need to introduce another init (private) that takes it. 

It's weird that one receives the `T` and the other `Option[T]`.

Reply:
hmm, I guess the only reason the other one is taking an `Option[T]` is because of the `ShardedDaemonProcessSettings`.

Reply:
Yes, I think we can keep it in the "power user" init method.